internal/handlers/common/filter.go

```diff
@@ -312,15 +312,18 @@ func filterFieldExpr(doc *types.Document, filterKey string, expr *types.Document
 			if !ok {
 				return false, NewErrorMsg(ErrBadValue, "$in needs an array")
 			}
+
 			var found bool
 			for i := 0; i < arr.Len(); i++ {
-				arrValue := must.NotFail(arr.Get(i))
-				switch arrValue := arrValue.(type) {
+				if found {
+					break
+				}
+
+				switch arrValue := must.NotFail(arr.Get(i)).(type) {
 				case *types.Array:
 					fieldValue, ok := fieldValue.(*types.Array)
 					if ok && matchArrays(fieldValue, arrValue) {
 						found = true
-						break
```

Comment:
That was a bug… kinda. This `break` terminates `switch`, not a `for` loop

Reply:
ah, see